controllers/index.js

```diff
@@ -48,6 +47,27 @@ exports.index = (req, res) => {
         }
     }
 
+    // in a reverse proxy situation we have to use the referer to retrieve
+    // the correct protocol, hostname, and path
+    // unfortunately this won't work, when accessng the page directly:
+    // e.g.: http://example.com/hacktoberfest/?username=XXX
+    // in such a case we set hostname to an empty string and create the link
+    // with js after the page has loaded
+    if (req.headers['x-forwarded-for']) {
+        //console.log(JSON.stringify(req.headers));
+        const referer = req.headers.referer;
+        if (referer) {
+            var hostname = referer.split("?")[0].slice(0, -1);
+            if (hostname.endsWith('/m')) {
+                hostname = hostname.slice(0, -2);
+            }
```

Comment:
Can you just explain what' going on here? Will this still work if not hosted on a .com?

Reply:
Yes, it will still work on a .com domain.

This is a rather interesting code path:

First it checks the referer and creates the hostname from it. This would be the flow when clicking the `Check` button. (Usually the hostname would also have the trailing slash, but since you already used the variable hostname without the slash, I also removed the trailing slash.)

However, if you click on the `.../me`link or browse directly to the `.../me` link, the referer will be the `.../me` link (due to the redirectin your app). In the step above I removed the last character, because I assumed it's the trailing slash. In case of the `.../me` link that's not the case, thus I have to check for `/m` instead of for `/me` and then remove the last 2 characters to get the hostname.

Then the last possibility is, if a user directly accesses the page `https://example.com/subdir/?user=XXX`, in which case there won't be a referer at all.
I create an empty hostname, so that I can apply the last workaround after the page has loaded.

This entire code path is only used when in a reverse proxy setup.

I could have written it differently, but it would have had more lines and more tests/checks. I'm a performance guy. The less instructions, the better.

Reply:
I guess not a big issue as only done with the x-forwarded-for set but this seems a little flaky to me, i.e. if we wanted to change the /me url to be something like /yourstats for example, this would break? I don't have any better solutions as I've not really experienced the issues you have behind a reverse proxy.

Reply:
Yes, in that case it would break. But if you wanted to change `/me` to `/yourstats`, you would have to change the `/me` code in several places, so I don't think this one line would make a difference. 

Also, please try to run it behind a reverse proxy and not in the root of the virtual host. Then your entire app will break. (e.g. `http://localhost/subdir/` or `https://hacktoberfestchecker.herokuapp.com/checker/`) 

It's clear that you haven't experienced these issues, because you are running it in the root of  a virtual host. But you can't expect that from all of your users.

Reply:
I'm running it [here](https://evermeet.cx/hfc/) and it works perfectly. I also tested it without reverse proxy and when running in the root of a virtual host. It doesn't change anything for your setup. It only makes it location agnostic thus you can now run your app on:

`https://example.com`
`https://example.com/hacktoberfest/`
`https://example.com/apps/hfc/`
and so on.

Without my change, your app can only be run as `https://example.com`.

Reply:
Good responses :) off to bed now but will take a closer look in the morning, in the mean time are you able to fix the build?

Reply:
eslint makes a few errors. this is not my fault. But I can explain this tomorrow, if you want.

Btw, I will make a small change, so that future changes to `/me` won't break the code. You only set a var and that's it.